controller/rest/sigstore.go

```diff
@@ -0,0 +1,379 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/neuvector/neuvector/controller/api"
+	"github.com/neuvector/neuvector/share"
+)
+
+func handlerSigstoreRootOfTrustPost(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	body, _ := ioutil.ReadAll(r.Body)
+	var rootOfTrust api.RESTSigstoreRootOfTrust
+	err := json.Unmarshal(body, &rootOfTrust)
+	if err != nil {
+		msg := fmt.Sprintf("could not unmarshal request body: %s", err.Error())
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, msg)
+		return
+	}
+
+	if *rootOfTrust.Name == "" {
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, "Field \"name\" cannot be empty.")
+		return
+	}
+
+	clusRootOfTrust := share.CLUSSigstoreRootOfTrust{
+		Name:           *rootOfTrust.Name,
+		RekorPublicKey: *rootOfTrust.RekorPublicKey,
+		RootCert:       *rootOfTrust.RootCert,
+		SCTPublicKey:   *rootOfTrust.SCTPublicKey,
+	}
+
+	err = clusHelper.PutSigstoreRootOfTrust(&clusRootOfTrust)
+	if err != nil {
+		msg := fmt.Sprintf("could not save root of trust to kv store: %s", err.Error())
+		restRespErrorMessage(w, http.StatusInternalServerError, api.RESTErrFailWriteCluster, msg)
+		return
+	}
+
+	msg := fmt.Sprintf("Added verifier \"%s\"", clusRootOfTrust.Name)
+	restRespSuccess(w, r, nil, nil, nil, nil, msg)
+}
+
+func handlerSigstoreRootOfTrustGetByName(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	rootName := ps.ByName("root_name")
+	rootOfTrust, err := clusHelper.GetSigstoreRootOfTrust(rootName)
+	if err != nil {
+		restRespErrorMessage(w, http.StatusInternalServerError, api.RESTErrFailReadCluster, err.Error())
+		return
+	}
+	if rootOfTrust == nil {
+		restRespError(w, http.StatusNotFound, api.RESTErrNotFound)
+	}
+	resp := CLUSRootToRESTRoot(rootOfTrust)
+	if withVerifiers(r) {
+		verifiers, err := clusHelper.GetAllSigstoreVerifiersForRoot(rootName)
```

Comment:
should it get from cache? also for line 131, 258

Reply:
Ideally, yes. That implementation will happen in another PR.